keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,350 @@
+overly restrictive and against the spirit of community-friendly, safe API evolution which is so important
+to a large project like Kubernetes with many external functional plugins.
+
+Ideally, a fix here would “just work” for “back-rev” implementations of NetworkPolicy 
+(which are out-of-tree - generally implemented by cni plugins) and would not 
+introduce new surprises for users. But in practicality this is not the case 
+because, by adding a new selection data structure, the logical inclusion/exclusion 
+properties of namespace and pod selectors become drastically different, as shown in the table below:
+
+
+In this table "policy-reader" might be something like the "calico-controller" or any other cni agent
+that reads the API Servers network policy portfolio and responds to it over time.
+
+|         input policy         | policy-reader (old)    | client (new)        |   |   |
+|------------------------------|------------------------|---------------------|---|---|
+| nsNameSelector + podSelector | ANY ns, pod restricted | ns + pod restricted |   |   |
+| nsNameSelector               | invalid (empty peer)   | ns restricted       |   |   |
+| podSelector                  | ANY namespace          | ANY namespace       |   |   |
+
+In an older client (one which predated the addition of a new, "nsNameSelector" field, we find
+that there is a scenario (`ANY ns, pod restricted`), wherein an old interpretation of the API
+assumes that the ONLY restricting field is a pod label.  Thus, although the user intended to
+restrict all traffic to a namespace/pod combination, the policy implementer does something **much**
+more promiscuous.
+
+This is all avoidable if we could assume, in all cases, that namespaces have a default label.
+
+There may also be many other usecases where the a security or permissions boundary might be conveniently defined in terms
+of a namepsace name, and that such a definition would be most easily referenced by a default label.
+
+### Goals
+
+- Add an immutable default label to ALL namespaces, as the namespace name so this 
+can be used as selector by arbitrary components.
+
+### Non-Goals
+
+* Publishing name labels on all resources
+* Exposing arbitrary fields as selectors
+
+## Proposal
+
+We propose to label of all namespaces, by default, with a reserved label 
+(i.e. “kubernetes.io/metadata.name” or some such), to allow easy selection.
+
+If this label is missing, it is added by the apiserver
+
+If the label is incorrect or changed, the apiserver fails to validate the object
+
+
+### User Stories (Optional)
+
+See https://github.com/kubernetes/enhancements/pull/2113/files for relevant user stories.
+
+### Notes/Constraints/Caveats (Optional)
+
+
+### Risks and Mitigations
+
+* Every namespace gets bigger in size. We think this is inconsequential
+* Some user may already be using that label already. This is mitigated by the 
+fact that the whole "kubernetes.io" label prefix is reserved. However, since there
```

Comment:
Find the link to where we reserved them all.  These are pretty clearly "ours" and I remember writing a doc claiming them and a large chuck of namespaces at the same time.

Reply:
will update the doc with this link https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ which explicitly calls this out.